sleap_io/io/nwb.py

```diff
@@ -6,27 +6,298 @@
+                name=training_frame_name,
+                description=training_frame_annotator,
+                unit="NA",
+                format="external",
+                external_file=[training_frame_video.filename],
+                dimension=[
+                    training_frame_video.shape[1],
+                    training_frame_video.shape[2],
+                ],
+                starting_frame=[0],
+                rate=30.0, # change to `video.backend.fps` when available
+            )
+        training_frame = TrainingFrame(
+            name=training_frame_name,
+            annotator=training_frame_annotator,
+            skeleton_instances=training_frame_skeleton_instances,
+            source_video=source_video,
+            source_video_frame_index=training_frame_video_index,
+        )
+        training_frame_list.append(training_frame)
+
+    training_frames = TrainingFrames(training_frames=training_frame_list)
+    pose_training = PoseTraining(
+        training_frames=training_frames,
+        source_videos=videos_to_source_videos(labels.videos),
+    )
+    return pose_training
+
+
+def slp_skeleton_to_nwb(skeleton: SLEAPSkeleton) -> NWBSkeleton:  # type: ignore[return]
+    """Converts SLEAP skeleton to NWB skeleton.
+
+    Args:
+        skeleton: A SLEAP skeleton.
+
+    Returns:
+        An NWB skeleton.
+    """
+    nwb_edges: list[list[int, int]]
+    if not isinstance(skeleton, SLEAPSkeleton):
+        raise ValueError("The input must be a SLEAP Skeleton object.")
+
+    skeleton_edges = {i: node for i, node in enumerate(skeleton.nodes)}
+    nwb_edges = []
+    for i, source in skeleton_edges.items():
+        for destination in list(skeleton_edges.values())[i:]:
+            if Edge(source, destination) in skeleton.edges:
+                nwb_edges.append([i, list(skeleton_edges.values()).index(destination)])
+
+    if len(skeleton.nodes) == 1:
+        name = f"Node {skeleton.nodes[0].name}"
+    elif len(skeleton.nodes) == 2:
+        name = f"Nodes {skeleton.nodes[0].name}, {skeleton.nodes[1].name}"
+    else:
+        name = f"Nodes {skeleton.nodes[0].name}, ..., {skeleton.nodes[-1].name}"
+    return NWBSkeleton(
+        name=name,
+        nodes=skeleton.node_names,
+        edges=np.array(nwb_edges, dtype=np.uint8),
+    )
+
+
+def instance_to_skeleton_instance(instance: Instance) -> SkeletonInstance:  # type: ignore[return]
+    """Converts a SLEAP Instance to an NWB SkeletonInstance.
+
+    Args:
+        instance: A SLEAP Instance.
+
+    Returns:
+        An NWB SkeletonInstance.
+    """
+    if not isinstance(instance, Instance):
+        raise ValueError("The input must be a SLEAP Instance object.")
+    
+    skeleton = slp_skeleton_to_nwb(instance.skeleton)
+    points_list = list(instance.points.values())
+    node_locs = [[point.x, point.y] for point in points_list]
+    np_node_locations = np.array(node_locs)
+    return SkeletonInstance(
+        name=name_generator("skeleton_instance"),
```

Comment:
```suggestion
        name=f"skeleton_instance_{id}",
```
Name according to globally unique ID, which should be an input to this function.